src/generators/ast-js/index.mjs

```diff
@@ -0,0 +1,47 @@
+import { createJsLoader } from '../../loader.mjs';
+import { createJsParser } from '../../parser.mjs';
+
+/**
+ * This generator parses Javascript sources passed into the generator's input
+ * field. This is separate from the Markdown parsing step since it's not as
+ * commonly used and can take up a significant amount of memory.
+ *
+ * Putting this with the rest of the generators allows it to be lazily loaded
+ * so we're only parsing the Javascript sources when we need to.
+ *
+ * @typedef {unknown} Input
+ *
+ * @type {import('../types.d.ts').GeneratorMetadata<Input, Array<JsProgram>>}
+ */
+export default {
+  name: 'ast-js',
+
+  version: '1.0.0',
+
+  description: 'Parses Javascript source files passed into the input.',
+
+  dependsOn: 'ast',
+
+  /**
+   * @param {Input} _
+   * @param {Partial<GeneratorOptions>} options
+   */
+  async generate(_, options) {
+    const { loadFiles } = createJsLoader();
+
+    if (!options.input) {
```

Comment:
Rather have it also allow an empty input (?? []), in the end this will load nothing and execute nothing right?